backend/routes.ts

```diff
@@ -2,16 +2,43 @@ import { Hono } from "hono";
 import { v4 as uuidv4 } from "uuid";
 import { authMiddleware, type AuthContext } from "./middleware.ts";
 import { type Env } from "./types.ts";
+import { fetchRequestHandler } from "@trpc/server/adapters/fetch";
+import { appRouter } from "./trpc/index.ts";
 
 // Import unified API key routes
 import apiKeyRoutes from "./api-key-routes.ts";
 import {
   isUsingLocalProvider,
   validateProviderConfig,
 } from "./providers/api-key-factory.ts";
+import { extractAndValidateUser, ValidatedUser } from "./auth.ts";
+import { createPermissionsProvider } from "./notebook-permissions/factory.ts";
+import { TrcpContext } from "./trpc/trpc.ts";
 
 const api = new Hono<{ Bindings: Env; Variables: AuthContext }>();
 
+// tRPC endpoint
+api.all("/trpc/*", async (ctx) => {
+  return fetchRequestHandler({
```

Comment:
I think since this is a fetch request handler you can pull this up out of hobo so it doesn't have to have two layers of routing, like I did with /graphql.